pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -298,19 +296,21 @@ def _generate_and_preprocess_model_data(  # type: ignore
                 self.X[self.date_column].iloc[1] - self.X[self.date_column].iloc[0]
             ).days
 
-    def _save_input_params(self, idata) -> None:
-        """Saves input parameters to the attrs of idata."""
-        idata.attrs["date_column"] = json.dumps(self.date_column)
-        idata.attrs["adstock"] = json.dumps(self.adstock.lookup_name)
-        idata.attrs["saturation"] = json.dumps(self.saturation.lookup_name)
-        idata.attrs["adstock_first"] = json.dumps(self.adstock_first)
-        idata.attrs["control_columns"] = json.dumps(self.control_columns)
-        idata.attrs["channel_columns"] = json.dumps(self.channel_columns)
-        idata.attrs["adstock_max_lag"] = json.dumps(self.adstock.l_max)
-        idata.attrs["validate_data"] = json.dumps(self.validate_data)
-        idata.attrs["yearly_seasonality"] = json.dumps(self.yearly_seasonality)
-        idata.attrs["time_varying_intercept"] = json.dumps(self.time_varying_intercept)
-        idata.attrs["time_varying_media"] = json.dumps(self.time_varying_media)
+    def create_idata_attrs(self) -> dict[str, str]:
+        attrs = super().create_idata_attrs()
+        attrs["date_column"] = json.dumps(self.date_column)
+        attrs["adstock"] = json.dumps(self.adstock.lookup_name)
+        attrs["saturation"] = json.dumps(self.saturation.lookup_name)
+        attrs["adstock_first"] = json.dumps(self.adstock_first)
+        attrs["control_columns"] = json.dumps(self.control_columns)
+        attrs["channel_columns"] = json.dumps(self.channel_columns)
+        attrs["adstock_max_lag"] = json.dumps(self.adstock.l_max)
+        attrs["validate_data"] = json.dumps(self.validate_data)
+        attrs["yearly_seasonality"] = json.dumps(self.yearly_seasonality)
+        attrs["time_varying_intercept"] = json.dumps(self.time_varying_intercept)
+        attrs["time_varying_media"] = json.dumps(self.time_varying_media)
```

Comment:
I wonder if we could store these attributes in a configuration yml file so that we can add or remove one we just do it in one place and no need to manually check

Reply:
Are you saying yml file *instead* of the InferenceData attrs?

Reply:
No, in addition. I'm open to ideas ;) Just a centralized object that carries all these names. Imagine that we need to add another field. Ideally we just add it in one place and it just works.

Reply:
Wouldnt we have to always write parsing to and parsing from the current attrs / dict, json? Have a small example in mind?

Reply:
yes! you are probably right! Then we need to make sure our unit-tests are good to help us with additional updates and changes :)